presto-native-execution/presto_cpp/main/runtime-metrics/prometheus-reporter/PrometheusReporter.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "presto_cpp/main/common/Configs.h"
+#include "velox/common/base/Exceptions.h"
+#include "velox/velox/common/base/GTestMacros.h"
+#include "velox/velox/common/base/StatsReporter.h"
+
+namespace facebook::presto::prometheus {
+
+struct StatsInfo {
+  velox::StatType statType;
+  void* metricPtr;
+};
+
+/// Prometheus CPP library exposes following classes:
+/// 1. Registry.
+/// 2. Family.
+/// 3. Specific metric type classes like Counter, Gauge, Histogram etc.
+///
+/// A family of metrics will have the same metric name. For example
+/// http_latency_ms. Different members of a family will have unique labels.
+/// For metric http_latency_ms labels could be {method="GET"}, {method="PUT"},
+/// {method="POST"} etc. Prometheus treats {<metric_name>, [labels]} as unique
+/// metric object.
+class PrometheusReporter : public facebook::velox::BaseStatsReporter {
+  class Impl;
```

Comment:
Can we give it a more meaningful name? And declare the class contents in header file, too?